File: API/Entities/Product.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace API.Entities;

[Table("Products")]
public class Product
{
    public int Id { get; set; }
    public string Name { get; set; }
    public int Price { get; set; }
    public string PhotoUrl { get; set; }
    public List<Photo> Photos { get; set; } = new();
    public List<Material> Materials { get; set; }
    public DateTime Created { get; set; } = DateTime.UtcNow;
    public DateTime Started { get; set; }
    public DateTime Ended { get; set; }
    public AppUser Employee { get; set; }
    public Category Category { get; set; }
}

File: UI/src/app/models/product.ts
import { Category } from './category';
import { Material } from './material';
import { Photo } from './photo';
import { User } from './user';

export interface Product {
  id: number;
  name: string;
  photoUrl: string;
  price: number;
  category: Category;
  photos: Photo[];
  materials: Material[];
  created: Date;
  started?: Date;
  ended?: Date;
  employee: User;
}
